.github/workflows/build.yml

```diff
@@ -77,25 +82,35 @@ jobs:
                   build_doc = "false"
           oses = ["macos", "ubuntu", "windows"]
           build_dict = {os : [k for k in build if k.startswith(os)] for os in oses}
-          print(f"::set-output name=build::{build_dict}")
-          print(f"::set-output name=test::{dict({os : [k for k in test if k.startswith(os)] for os in oses})}")
-          print(f"::set-output name=build_doc::{build_doc}")
-          for os in oses:
-              print(f"::set-output name=do_{os}::{'true' if len(build_dict[os]) > 0 else 'false'}")
-          print(f"::set-output name=python_version::{python_version}")
+
+          with open(environ["GITHUB_OUTPUT"], "a") as f:
+              f.write(f"build={dumps(build_dict)}\n")
+              f.write(f"test={dumps(dict({os : [k for k in test if k.startswith(os)] for os in oses}))}\n")
+              f.write(f"build_doc={dumps(build_doc)}\n")
```

Comment:
You are modifying the contents of these variables without giving any reason; to get the same values, dumps() should be called only for "build" and "test".